### ISSUE TITLE
VST3: adding accessibility to the generic vst interface

### ISSUE THREAD

**Issue by king-dahmanus** [audacity/audacity#2313]
Hello audacity developement team. I'd like to suggest a feature related to accessibility for blind people who use screen readers. Please add accessibility to the generic vst interface so we can select presets, and ajust sliders of the different effects. I need this feature to be able to customize the effect I want to apply and use the functionality of the plugin(s) to its full potencial. Hope you consider my request. Thanks, and have a great time

**Comment by DavidBailes** [comment]
There currently is an option to display the controls of a vst effect as a series of sliders, rather than an inaccessible gui. To change the display option for a vst effect:
1. open the vst effect.
2. tab to the manage button and press it. A pop-up menu opens.
3. Choose options on the menu. A vst effect options dialog opens.
4. tab to the enable graphical user interface check box, and uncheck it.
5. Close the vst effect options dialog, and close the vst effect dialog.
6. Re open the vst effect dialog, and there will now be a series of sliders for the parameters.

Notes on using these sliders with a screen reader. When you change the value of a slider, it just reads a number between 0 and 100. To find out the corresponding change in the parameter, you have to read the name of the slider again by pressing the keystroke used by your screen reader to read the current line. If you're using jaws, you have to tab away from the control and back to it so that it reads the correct value. You don't have to do this if you're using NVDA or narrator.

Concerning presets, see the first few commands on the menu which opens when you press the manage button.

**Comment by king-dahmanus** [comment]
thanks for your help, it works now. I really didn't think that would ever
exist. Thanks for your great help

**Comment by LWinterberg** [comment]
confirmed that the plain UI is present in the upcoming VST3 as well, so this is in across the board :)

**Comment by DavidBailes** [comment]
I don't think this should be closed until someone has tested this with a screen reader. In particular, how any sliders are read needs to be checked.
I've just installed the VST version of this plugin:
https://www.voxengo.com/product/tubeamp/
Tried to test using the windows 64 bit build from here:
https://github.com/audacity/audacity/actions/runs/1728897264

On opening the effect dialog, on the manage menu "options..." is greyed out, so I couldn't change to plain UI to do any testing.

**Comment by vsverchinsky** [comment]
> I don't think this should be closed until someone has tested this with a screen reader. In particular, how any sliders are read needs to be checked. I've just installed the VST version of this plugin: https://www.voxengo.com/product/tubeamp/ Tried to test using the windows 64 bit build from here: https://github.com/audacity/audacity/actions/runs/1728897264
> 
> On opening the effect dialog, on the manage menu "options..." is greyed out, so I couldn't change to plain UI to do any testing.

Will add this option soon, in the upcoming PR update. Though some plugins have tons of preferences and we may yet to figure out how to layout them better.

**Comment by king-dahmanus** [comment]
thanks for your help. I'll test it whenever the other forc I'm using gets
that pr update.

**Comment by DavidBailes** [comment]
I've tested the accessibility of the plain gui for vst3 effects, and it's not accessible for users of screen readers. However fixing this needs to wait until the following is fixed:
https://github.com/audacity/audacity/issues/2555
as fixing that may change which controls are used in the plain gui.

As it stands there are problems with the names of two types of control not being read, units not being read, and the absolute values of the parameters which use sliders not being read and indeed not being displayed for sighted keyboard users.

**Comment by LWinterberg** [comment]
Names and values are both displayed and read out to me when using the generic interface, so this appears fixed.
